liquid-compiler/src/grammar.pest

```diff
@@ -0,0 +1,56 @@
+WHITESPACE = _{" "}
+LINE_OR_SPACE = _{ NEWLINE | WHITESPACE }
+LiquidFile = ${ SOI ~ Element* ~ EOI }
+
+// Element-level parsing
+Element = _{ Expression | Tag | Raw } 
+
+TagStart = _{ (LINE_OR_SPACE* ~ "{%-") | "{%" }
+TagEnd =   _{ ("-%}" ~ LINE_OR_SPACE*) | "%}" }
+ExpressionStart = _{ (LINE_OR_SPACE* ~ "{{-") | "{{" }
+ExpressionEnd =   _{ ("-}}" ~ LINE_OR_SPACE*) | "}}" }
+
+Tag = !{ TagStart ~ Identifier ~ TagToken* ~ TagEnd }
```

Comment:
Without a pluggable parser of some sort, we end up requiring a catch-all `TagToken` that knows about every tags possible set of tokens and then we manually process those tokens later.

Reply:
> Not finding it but I thought I remember seeing a comment about preserving `{{ value }}` being treated differently than `{{ value | filter }}`. I don't know if I'd worry about that myself.

Probably you're refering to the `grammar.pest` comments. That was a suggestion I made in an earlier phase so `Value` and `Filterchain` would be parsed differently on TagToken, so one wouldn't "eat" the other when parsing. But right now `Filterchain` "eats" everything and `TagTokens` takes care of "unwrapping" the value out of the filterchain so I think there won't be a problem anymore.